cmd/spire-server/cli/federation/common.go

```diff
@@ -1,9 +1,160 @@
 package federation
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
 	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"github.com/spiffe/spire/cmd/spire-server/util"
 )
 
+// FederationRelationship type is used for parsing federation relationships from file
+type federationRelationshipJSON struct {
+	TrustDomain           string          `json:"trust_domain,omitempty"`
+	BundleEndpointURL     string          `json:"bundle_endpoint_url,omitempty"`
+	BundleEndpointProfile string          `json:"bundle_endpoint_profile,omitempty"`
+	EndpointSPIFFEID      string          `json:"endpoint_spiffe_id,omitempty"`
+	Bundle                json.RawMessage `json:"bundle,omitempty"`
+	BundleFormat          string          `json:"bundle_format,omitempty"`
+	BundlePath            string
+}
+
+// federationRelationshipsFromFile parse a json file into types FederationRelationships
+func federationRelationshipsFromFile(path string) ([]*types.FederationRelationship, error) {
+	r := os.Stdin
+	if path != "-" {
+		f, err := os.Open(path)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+		r = f
+	}
+
+	dat, err := io.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	relationships := &federationRelationships{}
+	if err := json.Unmarshal(dat, &relationships); err != nil {
+		return nil, fmt.Errorf("failed to parse JSON: %w", err)
+	}
+
+	protoRelationships := []*types.FederationRelationship{}
+	for i, relationship := range relationships.FederationRelationships {
+		protoRelationship, err := jsonToProto(relationship)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse item %d: %w", i, err)
+		}
+		protoRelationships = append(protoRelationships, protoRelationship)
+	}
+
+	return protoRelationships, nil
+}
+
+func jsonToProto(fr *federationRelationshipJSON) (*types.FederationRelationship, error) {
+	if fr.TrustDomain == "" {
+		return nil, errors.New("trust domain is required")
+	}
+
+	if fr.BundleEndpointURL == "" {
+		return nil, errors.New("bundle endpoint URL is required")
+	}
+
+	proto := &types.FederationRelationship{
+		TrustDomain:       fr.TrustDomain,
+		BundleEndpointUrl: fr.BundleEndpointURL,
+	}
+
+	switch fr.BundleEndpointProfile {
+	case profileHTTPSWeb:
+		if fr.EndpointSPIFFEID != "" {
```

Comment:
I'm not sure how much of this extra validation we should do here in the CLI. I think we should probably rely on the API.